src/Language/PureScript/Parser/Lexer.hs

```diff
@@ -173,6 +174,12 @@ updatePositions (x:xs) = x : zipWith update (x:xs) xs
 parseTokens :: Lexer u [PositionedToken]
 parseTokens = whitespace *> P.many parsePositionedToken <* P.skipMany parseComment <* P.eof
 
+lexLenient :: FilePath -> Text -> Either P.ParseError [PositionedToken]
+lexLenient f s = updatePositions <$> P.parse parseTokensLenient f s
+
+parseTokensLenient :: Lexer u [PositionedToken]
```

Comment:
I just meant that you could maybe add a copy of an existing `psc-ide` test which has a syntax error after the imports somewhere. Does that work?